### PR TITLE
Fix multiple connection acquiring and lock release error

### PR DESCRIPTION
**Pull request by DriverX**
Fix multiple lock acquire in `ConnectionsPool.acquire` and other places where using `asyncio.Condition` with custom `aioredis.locks.Lock`

see https://bugs.python.org/issue32734

### CODE REVIEW

File: tests/pool_test.py [hunk 1]
```diff
@@ -542,3 +543,44 @@ async def test_pool__drop_closed(create_pool, server):
...
+
+        with pytest.raises(asyncio.CancelledError):
+            await conn_fut1
+        with pytest.raises(asyncio.CancelledError):
+            await conn_fut2
+
+        await conn_fut3
+
+        # did not raise!!
+        # with pytest.raises(RuntimeError, match='Lock is not acquired'):
```
Comment (seandstewart) [inline tests/pool_test.py:585]: @DriverX - do we expect this error to be raised?

**Comment by seandstewart** [comment]
@DriverX  - thank you for your submission!

Just one question remaining around your test, and then if you can sign the CLA that would be much appreciated!

File: tests/pool_test.py (hunk 1 above)
Comment (DriverX) [reply tests/pool_test.py:570]: nope. I'll remove this comment

**Comment by seandstewart** [comment]
> > @DriverX - thank you for your submission!
> > Just one question remaining around your test, and then if you can sign the CLA that would be much appreciated!
> 
> Sorry. My commit have not email and i can't sign CLA for this PR.

That's alright, are you still able to sign with your GH account?

**Review by seandstewart — APPROVED** [review]

**Pull request merged**